gitlabform/processors/project/schedules_processor.py

```diff
@@ -42,102 +48,124 @@ def _process_configuration(self, project_and_group: str, configuration: dict):
                     debug(
                         "Changing existing pipeline schedule '%s'", schedule_description
                     )
```

Comment:
The latest logic of `_update_existing_schedule` is performing checks to see if update is needed. This log should be in there instead. Otherwise the log might say "Changing schedule..." but actually that might not be happening.

Another option, move the logic here to see if update is needed and then call the update method. Maybe we don't even need the `_update_existing_schedule` function since the logic is not that extensive.